Integrations/integration-Remedy-On-Demand.yml

```diff
@@ -0,0 +1,448 @@
+detaileddescription: Use Remedy On-Demand to manage tickets
+configuration:
+- display: Server Url (e.g. 'https://myurl.com', 'http://41.79.151.82')
+  name: url
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Port
+  name: port
+  defaultvalue: "8008"
+  type: 0
+  required: true
+- display: Username
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+script:
+  script: |
+    // ensure '/' at the end of the url
+    if (params.url[params.url.length - 1] === '/') {
+        params.url = params.url.substring(0, params.url.length - 1);
+    }
+
+    var baseUrl = params.url + ':' + params.port + '/';
+
+    var stringify = function(obj) {
+        return JSON.stringify(obj, null, 2);
+    }
+
+    // returns incId padded with '0's that is 15 length string. e.g. '82' -> '000000000000082'
+    var preperIncId = function (incId) {
+        var res = '000000000000000' + incId;
+        return res.substr(-15);
+    }
+
+    var createTableEntry = function (name, contents, context, headers) {
+        return {
+            // type
+            Type: entryTypes.note,
+             // contents
+            ContentsFormat: formats.json,
+            Contents: contents,
+            // human-readable
+            ReadableContentsFormat: formats.markdown,
+            HumanReadable: tableToMarkdown(name, contents, headers),
+            // context
+            EntryContext: context
+        };
+    }
+
+    // mutetor that removes all falsly fields
+    var filterEmptyFields = function(obj) {
+        Object.keys(obj).forEach(function(key) { !obj[key] && delete obj[key]; });
+    }
+
+    var sendRequest = function(url, token, method, body) {
+
+        var res = http(
+            url,
+            {
+                Method: method || 'GET',
+                Headers: {
+                    'Content-Type': ['application/json'],
+                    'Authorization': ['AR-JWT ' + token]
+                },
+                Body: body
+            }
+        );
+
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed'
+                + '\nurl: ' + url
+                + '\nStatus code: ' + res.StatusCode
+                + '\nBody: ' + JSON.stringify(res);
+        }
+
+        return res;
+    };
+
+    var login = function() {
+        var url = baseUrl + 'api/jwt/login/';
+
+        var body = {
+            username: params.credentials.identifier,
+            password: params.credentials.password
+        };
+
+        var res = http(
+            url,
+            {
+                Method: 'POST',
+                Headers: {
+                    'Content-Type': ['application/x-www-form-urlencoded']
+                },
+                Body: encodeToURLQuery(body).replace(/^\?/, '')
```

Comment:
if you use regex here, also please use it in line 28 mutating the address.
Either use string mutators or regex, but using both is confusing